Diagnostic/Providers/Builtin.py

```diff
@@ -0,0 +1,183 @@
+    'disk':       [ 'readspersecond', 'writespersecond', 'transferspersecond', 'averagereadtime', 'averagewritetime',
+                    'averagetransfertime', 'averagediskqueuelength', 'readbytespersecond', 'writebytespersecond',
+                    'bytespersecond' ]
+    }
+
+_omiClassName = { 'processor': 'SCX_ProcessorStatisticalInformation',
+                  'memory': 'SCX_MemoryStatisticalInformation',
+                  'network': 'SCX_EthernetPortStatistics',
+                  'filesystem': 'SCX_FileSystemStatisticalInformation',
+                  'disk': 'SCX_DiskDriveStatisticalInformation'
+                }
+
+_instancedClasses = [ 'network', 'filesystem', 'disk', 'processor' ]
+
+# The Azure Metrics infrastructure, along with App Insights, requires that quantities be measured
+# in one of these units: Percent, Count, Seconds, Milliseconds, Bytes, BytesPerSecond, CountPerSecond
+#
+# Some of the built-in metrics are retrieved in some other unit (e.g. "MiB") and need to be scaled
+# to the expected unit before being passed along the pipeline. The _scaling map holds all counterSpecifier
+# names that need to be scaled. If a counterSpecifier isn't in this list, no scaling is needed.
+_scaling = defaultdict(lambda:defaultdict(str),
+            { 'memory' : defaultdict(str,
+                { 'availablememory': 'scaleUp="1048576"',
+                  'usedmemory' : 'scaleUp="1048576"',
+                  'availableswap' : 'scaleUp="1048576"',
+                  'usedswap' : 'scaleUp="1048576"'
+                } )
+            } )
+
+_metrics = defaultdict(list)
+_eventNames = {}
+
+_defaultSampleRate = 15
+
+def SetDefaultSampleRate(rate):
+    _defaultSampleRate = rate
+
+class BuiltinMetric:
+    def __init__(self, counterSpec):
+        t = ProvUtil.GetCounterSetting(counterSpec, 'type')
+        if t is None:
+            self._Type = 'builtin'
+        else:
+            self._Type = t.lower()
+            if t != 'builtin':
+                raise ProvUtil.UnexpectedCounterType('Expected type "builtin" but saw type "{0}"'.format(self.Type))
+
+        self._CounterClass = ProvUtil.GetCounterSetting(counterSpec, 'class').lower()
+        if self._CounterClass not in _builtIns:
+            raise ProvUtil.InvalidCounterSpecification('Unknown Builtin class {0}'.format(self._CounterClass))
+        self._Counter = ProvUtil.GetCounterSetting(counterSpec, 'counter').lower()
+        if self._Counter not in _builtIns[self._CounterClass]:
+            raise ProvUtil.InvalidCounterSpecification(
+                'Counter {0} not in builtin class {1}'.format(self._Counter, self._CounterClass))
+        self._InstanceId = ProvUtil.GetCounterSetting(counterSpec, 'instanceId')
+        self._Label = ProvUtil.GetCounterSetting(counterSpec, 'counterSpecifier')
+        self._SampleRate = ProvUtil.GetCounterSetting(counterSpec, 'sampleRate')
+
+    def IsType(self, t):
+        return self.Type == t.lower()
```

Comment:
One underscore is private-by-convention. Two underscores is implementation-reserved.